recipes/iceoryx/all/conanfile.py

```diff
@@ -0,0 +1,244 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+import textwrap
+
+class IceoryxConan(ConanFile):
+
+    name = "iceoryx"
+    version = "1.0.0"
+    license = "Apache-2.0"
+    homepage = "https://iceoryx.io/"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Eclipse iceoryx - true zero-copy inter-process-communication"
+    topics = ("Shared Memory", "IPC", "ROS", "Middleware")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+         "shared":          [True, False],
+         "toml_config":     [True, False]
+    }
+    default_options = {
+        "shared":           False,
```

Comment:
```suggestion
        "shared":           False,
```

No fPIC as option?

Reply:
that is a good point - if i remember well i raised an issue on iceoryx and discussed with them. but seems they already set this in their CMAKE-Files via POSITION_INDEPENDENT_CODE - and I was not 100% sure if it is a "must" for CCI - since the hookups didnt show red lines ^^' so my basic questions here is : is it a must if the library itself / the project already set everything in their CMakefiles? That is now sth i will follow Conan community wishes.

Reply:
`fPIC` is only a must when the library is supposed to be used in a shared library or really required by some platform.
It looks like `POSITION_INDEPENDENT_CODE` is used all over the code: https://github.com/eclipse-iceoryx/iceoryx/search?p=1&q=POSITION_INDEPENDENT_CODE
It is possible to patch it, but I don't think it's worth the effort.

Reply:
Usually we do https://github.com/conan-io/conan-center-index/blob/41d4fbb8c218f9d938ccb8b9d888bff52002e62e/recipes/glbinding/all/conanfile.py#L42-L44 but the number of files....

Reply:
@prince-chrismc  i will give it a try - i never saw this so far - but cool it exist.